layouts/grid-bag.lisp

```diff
@@ -0,0 +1,143 @@
+(in-package #:org.shirakumo.alloy)
+
+(defclass grid-extent ()
+  ((grid-x
+   :initarg :col
+   :accessor col)
+  (grid-y
+   :initarg :row
+   :accessor row)
+  (grid-width
+   :initarg :width
+   :initform 0
+   :accessor width)
+  (grid-height
+   :initarg :height
+   :initform 0
+   :accessor height)))
+
+(defun make-grid-extent (col row &optional (width 1) (height 1))
+  (make-instance 'grid-extent
+                 :col col
+                 :row row
+                 :width width
+                 :height height))
+
+(defmethod grid-overlapping-p ((a grid-extent) (b grid-extent))
```

Comment:
This function does not look correct to me. For one, the `let*` does not even contain the `and`. Also, it should probably be a function rather than a generic function.